docs/conceptual/list.comparewith['t]-function-[fsharp].md

```diff
@@ -0,0 +1,71 @@
+---
+title: List.compareWith<'T> Function (F#)
+description: List.compareWith<'T> Function (F#)
+keywords: visual f#, f#, functional programming
+author: liboz
+---
+
+# List.compareWith<'T> Function (F#)
+
+Compares two lists using the given comparison function, element by element.
+
+**Namespace/Module Path**: Microsoft.FSharp.Collections.List
+
+**Assembly**: FSharp.Core (in FSharp.Core.dll)
+
+## Syntax
+
+```fsharp
+// Signature:
+List.compareWith : ('T -> 'T -> int) -> 'T list -> 'T list -> int
+
+// Usage:
+List.compareWith comparer source1 source2
+```
+
+#### Parameters
+*comparer*
+Type: **'T -&gt; 'T -&gt;**[int](https://msdn.microsoft.com/library/025d5455-3622-4ea5-9573-3ecbd4ee1375)
+
+A function that takes an element from each of the two source lists and returns an int. If it evaluates to a non-zero value iteration is stopped and that value is returned.
+
+*source1*
+Type: **'T** [list](https://msdn.microsoft.com/library/c627b668-477b-4409-91ed-06d7f1b3e4a7)
+
+The first input list.
+
+*source2*
+Type: **'T** [list](https://msdn.microsoft.com/library/c627b668-477b-4409-91ed-06d7f1b3e4a7)
+
+The second input list.
+
+## Return Value
+Returns the first non-zero result from the comparison function. If the end of a list is reached it returns a -1 if the first list is shorter and a 1 if the second list is shorter.
```

Comment:
@cartermp I changed the statement a bit more to be more accurate (mostly if the lengths are different the only possible return values are 1 and -1).